qutebrowser/browser/downloads.py

```diff
@@ -873,6 +926,12 @@ def remove_items(self, downloads):
                 download.deleteLater()
         self.endRemoveRows()
 
+    def update_indexes(self):
+        """Update indexes of all DownloadItems"""
+        for i, d in enumerate(self.downloads, 1):
+            if not d.index == i:
+                d.index = i
```

Comment:
Actually, ignore the previous comment - I'd just update the index unconditionally.